rclpy/src/rclpy/context.cpp

```diff
@@ -149,13 +152,20 @@ Context::shutdown()
 {
   {
     std::lock_guard guard{g_contexts_mutex};
-    g_contexts.erase(
-      std::remove(g_contexts.begin(), g_contexts.end(), rcl_context_.get()),
-      g_contexts.end());
+    if (std::find(g_contexts.begin(), g_contexts.end(), rcl_context_.get()) != g_contexts.end()) {
+      g_contexts.erase(
+        std::remove(g_contexts.begin(), g_contexts.end(), rcl_context_.get()),
```

Comment:
I think this searches for the context twice, and will keep searching the list even though the context should only be in it once. I'd recommend this instead

```c++
auto iter = std::find(....)
if (iter != g_contexts.end()) {
    g_contexts.erase(iter);
}
```

Reply:
nice point 👍 thanks!